dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/message/NettyRequestMessage.java

```diff
@@ -196,11 +199,52 @@ else if (host.length() > 1 && '/' == host.charAt(0) && '/' == host.charAt(1) &&
                 }
             }
             parseURLHost(host, start, slash_start);
+            parseURI(host.getBytes(), slash_start);
         }
         // Need to check if Scheme is also verified by Netty coded or add that ourselves
         // Probably need to add check of authority ourselves as well
     }
 
+    /**
+     * Parse the URI information out of the input data, along with any query
+     * information found afterwards. If format errors are found, then an
+     * exception is thrown.
+     *
+     * @param data
+     * @throws IllegalArgumentException
+     */
+    private void parseURI(byte[] data, int start) {
+        // at this point, we're parsing /URI [?querystring]
+        if (start >= data.length) {
+            // PK22096 - default to "/" if not found, should have caught empty
+            // string inputs previously (http://host:port is valid)
+            this.myURIBytes = SLASH;
+            if (tc.isDebugEnabled()) {
```

Comment:
TraceComponent.isAnyTracingEnabled()

Reply:
Interesting that this is the same logic used in legacy [here](https://github.com/OpenLiberty/open-liberty/blob/c8249ca16f2cb8e01cf5b098402ebedd7e7817f7/dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpRequestMessageImpl.java#L1165-L1167). I will clean this up and I created https://github.com/OpenLiberty/open-liberty/issues/32222 to follow up